### PR TITLE
User Story 208753 - App Service Backup

### PR DESCRIPTION
**Pull request by TomArcherMsft**
[User Story 208753](https://dev.azure.com/msft-skilling/Content/_workitems/edit/208753)

### CODE REVIEW

File: quickstart/101-app-service-backup/main.tf [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  services {
+    blob  = true
+    queue = false
+    table = false
+    file  = false
+  }
+
+  start  = "2024-01-01"
```
Comment (lonegunmanb) [inline quickstart/101-app-service-backup/main.tf]: Would this hard code date be invalid from next year? Could we try `formatdate("YYYY-MM-DD", timestamp())` for `start`, and `formatdate("YYYY-MM-DD", timeadd(timestamp(), "8765h"))` for `expire`, and add a comment to tell the reader, this dynamic date is only for testing purpose, please do not follow in the production environment?

File: quickstart/101-app-service-backup/main.tf [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+resource "azurerm_windows_web_app" "example" {
+  name                = random_string.windows_web_app_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  service_plan_id     = azurerm_service_plan.example.id
+
+  backup {
+    name                = "Example"
+    storage_account_url = "https://${azurerm_storage_account.example.name}.blob.core.windows.net/${azurerm_storage_container.example.name}${data.azurerm_storage_account_sas.example.sas}&sr=b"
```
Comment (lonegunmanb) [inline quickstart/101-app-service-backup/main.tf:108]: A question: I saw there's a question mark as delimiter in the url [here](https://learn.microsoft.com/en-us/azure/storage/common/storage-sas-overview#sas-token), do we need a delimiter after `${azurerm_storage_container.example.name}`?

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @TomArcherMsft for opening this pr! One comment and one question.

**Comment by stemaMSFT** [comment]
@TomArcherMsft are you still planning on working on this one? Saw it on our backlog.

File: quickstart/101-app-service-backup/main.tf (hunk 1 above)
Comment (TomArcherMsft) [reply quickstart/101-app-service-backup/main.tf]: Done

**Comment by TomArcherMsft** [comment]
@ms-henglu This PR comes from your batch of samples. Could you please reply to Zijie about the delimiter and let me know what I need to change? Thanks.

File: quickstart/101-app-service-backup/main.tf (hunk 2 above)
Comment (TomArcherMsft) [reply quickstart/101-app-service-backup/main.tf:108]: @ms-henglu This PR comes from your batch of samples. Could you please reply to Zijie about the delimiter and let me know what I need to change? Thanks.

**Comment by TomArcherMsft** [comment]
> @TomArcherMsft are you still planning on working on this one? Saw it on our backlog.

@stemaMSFT Yes, I'm working on it now. Thanks for checking!

File: quickstart/101-app-service-backup/main.tf (hunk 2 above)
Comment (ms-henglu) [reply quickstart/101-app-service-backup/main.tf:108]: Hi @TomArcherMsft , would you please share me the link of my samples? So I could help investigate. Thanks!

Comment (TomArcherMsft) [reply quickstart/101-app-service-backup/main.tf:108]: @ms-henglu https://github.com/ms-henglu/terraform-provider-azurerm/blob/master/examples/app-service/backup/main.tf

Comment (ms-henglu) [reply quickstart/101-app-service-backup/main.tf:108]: Thanks! I just confirmed that in the `data.azurerm_storage_account_sas.example.sas`, it has a prefix "?" . So I think we don't need a delimiter after ${azurerm_storage_container.example.name}

refs: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/vendor/github.com/hashicorp/go-azure-helpers/storage/sas_token.go#L64

**Comment by TomArcherMsft** [comment]
@stemaMSFT @lonegunmanb Can this PR be merged yet?

**Pull request merged**